lib/request.ts

```diff
@@ -0,0 +1,77 @@
+import * as typeis from 'type-is';
+import { InvalidArgumentError } from './errors';
+import { hasOwnProperty } from './utils/fn';
+
+export class Request {
+  body: any;
+  headers: any;
+  method: string;
+  query: any;
+  constructor(
+    options: {
+      body: any;
+      headers: any;
+      method: string;
+      query: any;
+      [key: string]: any;
+    } = {} as any,
+  ) {
+    if (!options.headers) {
+      throw new InvalidArgumentError('Missing parameter: `headers`');
+    }
+
+    if (!options.method) {
+      throw new InvalidArgumentError('Missing parameter: `method`');
+    }
+
+    if (typeof options.method !== 'string') {
```

Comment:
This fails linting:
~~~~
ERROR: .../node-oauth2-server/lib/request.ts:27:9 - Expression is always false.
~~~~

Question is a bit whether this check is really needed here, and couldn't just be dropped. After all, none of the others are checked in a similar way either.